models/repo.go

```diff
@@ -1583,14 +1584,24 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
 	}
 
 	repos = make([]*Repository, 0, opts.PageSize)
-
-	// Append conditions
-	sess := x.Where("LOWER(lower_name) LIKE ?", "%"+opts.Keyword+"%")
-	if opts.OwnerID > 0 {
-		sess.And("owner_id = ?", opts.OwnerID)
+	sess := x.Alias("r")
+	// Attempt to find repositories that opts.UserId has access to
+	// This does not include other people's private repositories even if opts.UserId is an admin
+	if !opts.Private && opts.UserID > 0 {
+		sess.Join("LEFT", []string{"team_repo", "tr"}, "tr.repo_id=r.id")
```

Comment:
>So you're saying the access table will contain a row for collaborators, and members of a team that has access to a private repository?

`access` table contains a record if:

1. Repo is not owned by the user
2. User has at least read access to the repo